client/client.go

```diff
@@ -1110,21 +1119,31 @@ func (c *Client) run() {
 	}
 }
 
+// nodeMapHashes returns the hashes of the passed Node's attribute and metadata
+// maps.
+func nodeMapHashes(node *structs.Node) (attrHash, metaHash uint64, err error) {
+	attrHash, err = hashstructure.Hash(node.Attributes, nil)
+	if err != nil {
+		return 0, 0, fmt.Errorf("unable to calculate node attributes hash: %v", err)
+	}
+	// Calculate node meta map hash
+	metaHash, err = hashstructure.Hash(node.Meta, nil)
+	if err != nil {
+		return 0, 0, fmt.Errorf("unable to calculate node meta hash: %v", err)
+	}
+	return attrHash, metaHash, nil
+}
+
 // hasNodeChanged calculates a hash for the node attributes- and meta map.
 // The new hash values are compared against the old (passed-in) hash values to
 // determine if the node properties have changed. It returns the new hash values
 // in case they are different from the old hash values.
 func (c *Client) hasNodeChanged(oldAttrHash uint64, oldMetaHash uint64) (bool, uint64, uint64) {
 	c.configLock.RLock()
 	defer c.configLock.RUnlock()
-	newAttrHash, err := hashstructure.Hash(c.config.Node.Attributes, nil)
-	if err != nil {
-		c.logger.Printf("[DEBUG] client: unable to calculate node attributes hash: %v", err)
-	}
-	// Calculate node meta map hash
-	newMetaHash, err := hashstructure.Hash(c.config.Node.Meta, nil)
+	newAttrHash, newMetaHash, err := nodeMapHashes(c.config.Node)
```

Comment:
This is the write copy of the config, no? A comment here stating that would be helpful for future debugging.